# Comparison of DevBrain IsoSeq vs CHESS Transcriptomes


## Download CHESS v 3.0.1 Datasets
### Primary Transcriptome
* 168451 transcripts

In [1]:
wget https://github.com/chess-genome/chess/releases/download/v.3.0.1/chess3.0.1.gtf.gz
gunzip chess3.0.1.gtf.gz

cat chess3.0.1.gtf | awk ' $3 == "transcript" { print }  ' | wc -l

--2024-05-29 05:50:56--  https://github.com/chess-genome/chess/releases/download/v.3.0.1/chess3.0.1.gtf.gz
Resolving github.com (github.com)... 140.82.116.4
Connecting to github.com (github.com)|140.82.116.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/128085164/ec49e4bb-ba11-4c9c-ba85-88582eb2342a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240529%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240529T125056Z&X-Amz-Expires=300&X-Amz-Signature=6335af706535712081ff0fcdb3ee4a80281d5919409a4b646169597600dc4a17&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=128085164&response-content-disposition=attachment%3B%20filename%3Dchess3.0.1.gtf.gz&response-content-type=application%2Foctet-stream [following]
--2024-05-29 05:50:57--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/128085164/ec49e4bb-ba11-4c9c-ba85-88582eb2342a

### Assembled Transcriptome 
* 987244 transcripts

In [3]:
wget https://github.com/chess-genome/chess/releases/download/v.3.0.1/assembled.gtf.gz
gunzip assembled.gtf.gz
cat assembled.gtf | awk ' $3 == "transcript" { print }  ' | wc -l


--2024-05-29 05:51:36--  https://github.com/chess-genome/chess/releases/download/v.3.0.1/assembled.gtf.gz
Resolving github.com (github.com)... 140.82.116.4
Connecting to github.com (github.com)|140.82.116.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/128085164/7e080e0c-348e-40a8-bc42-f7e36358b021?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240529%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240529T125136Z&X-Amz-Expires=300&X-Amz-Signature=61ddf983ef05d6e0819b16d427f38397cadba0a6773aab93b5811bf33f468c99&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=128085164&response-content-disposition=attachment%3B%20filename%3Dassembled.gtf.gz&response-content-type=application%2Foctet-stream [following]
--2024-05-29 05:51:36--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/128085164/7e080e0c-348e-40a8-bc42-f7e36358b021?X

## Download IsoSeq Data from Developing Human Brain
### hg19 version:
* 214516 transcripts (total)

In [4]:
wget https://github.com/gandallab/Dev_Brain_IsoSeq/raw/main/data/cp_vz_0.75_min_7_recovery_talon.gtf.gz
gunzip cp_vz_0.75_min_7_recovery_talon.gtf.gz
cat cp_vz_0.75_min_7_recovery_talon.gtf | awk ' $3 == "transcript" { print }  ' | wc -l


--2024-05-29 05:53:37--  https://github.com/gandallab/Dev_Brain_IsoSeq/raw/main/data/cp_vz_0.75_min_7_recovery_talon.gtf.gz
Resolving github.com (github.com)... 140.82.116.3
Connecting to github.com (github.com)|140.82.116.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/gandallab/Dev_Brain_IsoSeq/main/data/cp_vz_0.75_min_7_recovery_talon.gtf.gz [following]
--2024-05-29 05:53:37--  https://raw.githubusercontent.com/gandallab/Dev_Brain_IsoSeq/main/data/cp_vz_0.75_min_7_recovery_talon.gtf.gz
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 51639022 (49M) [application/octet-stream]
Saving to: ‘cp_vz_0.75_min_7_recovery_talon.gtf.gz’

100%[======================================>] 51,639,022   112MB

### CrossMap to hg38
* Install crossmap, download liftover chain

In [15]:
# mamba install CrossMap
wget https://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz
gunzip hg19ToHg38.over.chain.gz

--2024-05-29 06:06:43--  https://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz
Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163
Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 227698 (222K) [application/x-gzip]
Saving to: ‘hg19ToHg38.over.chain.gz.1’

100%[======================================>] 227,698     --.-K/s   in 0.03s   

2024-05-29 06:06:43 (6.89 MB/s) - ‘hg19ToHg38.over.chain.gz.1’ saved [227698/227698]



In [18]:
CrossMap  gff hg19ToHg38.over.chain cp_vz_0.75_min_7_recovery_talon.gtf devBrainIsoSeq_hg38.gtf

cat devBrainIsoSeq_hg38.gtf | awk ' $3 == "transcript" {print} ' | wc -l

cat devBrainIsoSeq_hg38.gtf | grep NOVEL > devBrainIsoSeq_hg38_novel.gtf

cat devBrainIsoSeq_hg38_novel.gtf | awk ' $3 == "transcript" {print} ' | wc -l

2024-05-29 06:07:20 [INFO]  Read the chain file "hg19ToHg38.over.chain" 
204191
141975


* devBrainIsoSeq_hg38.gtf has 204191 total transcripts
* devBrainIsoSeq_hg38_novel.gtf has 141975 novel transcripts


## Compare Transcriptome Annotations with GFFCOMPARE

In [ ]:
gffcompare --version

### 1. Isoseq vs Primary CHESS

In [20]:
gffcompare devBrainIsoSeq_hg38.gtf -r chess3.0.1.gtf -o IsoSeq_vs_Primary
cat IsoSeq_vs_Primary.stats

  168451 reference transcripts loaded.
  291 duplicate reference transcripts discarded.
  214394 query transfrags loaded.
# gffcompare v0.12.6 | Command line was:
#gffcompare devBrainIsoSeq_hg38.gtf -r chess3.0.1.gtf -o IsoSeq_vs_Primary
#

#= Summary for dataset: devBrainIsoSeq_hg38.gtf 
#     Query mRNAs :  214394 in   23376 loci  (206255 multi-exon transcripts)
#            (13767 multi-transcript loci, ~9.2 transcripts per locus)
# Reference mRNAs :  168160 in   61304 loci  (141572 multi-exon)
# Super-loci w/ reference transcripts:    19733
#-----------------| Sensitivity | Precision  |
        Base level:    40.9     |    62.8    |
        Exon level:    48.8     |    62.2    |
      Intron level:    53.6     |    79.5    |
Intron chain level:    31.6     |    21.7    |
  Transcript level:    28.4     |    22.3    |
       Locus level:    28.9     |    73.7    |

     Matching intron chains:   44805
       Matching transcripts:   47720
              Matching loci:   17687

       

* 47720 / 214394 IsoSeq transcripts (22%) are found in CHESS Primary GTF


### 2. Isoseq vs Assembled Transcripts

In [21]:
gffcompare -r assembled.gtf devBrainIsoSeq_hg38.gtf -o IsoSeq_vs_Assembly
cat IsoSeq_vs_Assembly.stats

  987244 reference transcripts loaded.
  115 duplicate reference transcripts discarded.
  214394 query transfrags loaded.
# gffcompare v0.12.6 | Command line was:
#gffcompare -r assembled.gtf devBrainIsoSeq_hg38.gtf -o IsoSeq_vs_Assembly
#

#= Summary for dataset: devBrainIsoSeq_hg38.gtf 
#     Query mRNAs :  214394 in   23376 loci  (206255 multi-exon transcripts)
#            (13767 multi-transcript loci, ~9.2 transcripts per locus)
# Reference mRNAs :  987129 in  168026 loci  (882562 multi-exon)
# Super-loci w/ reference transcripts:    16125
#-----------------| Sensitivity | Precision  |
        Base level:    12.8     |    77.8    |
        Exon level:    14.2     |    62.5    |
      Intron level:    24.1     |    84.4    |
Intron chain level:     6.5     |    27.8    |
  Transcript level:     5.9     |    27.0    |
       Locus level:     8.8     |    69.4    |

     Matching intron chains:   57263
       Matching transcripts:   57925
              Matching loci:   14792

       

* 57263 / 214394  IsoSeq transcripts (26.7%) are found in the Assembled 


### 3. IsoSeq vs (Primary + Assembled) Transcripts

In [24]:
gffcompare assembled.gtf chess3.0.1.gtf -o Primary_and_Assembled
gffcompare -r Primary_and_Assembled.combined.gtf devBrainIsoSeq_hg38.gtf -o IsoSeq_vs_Primary_and_Assembled
cat IsoSeq_vs_Primary_and_Assembled.stats

Loading query file #1: assembled.gtf
  987244 query transfrags loaded.
  868 duplicate query transfrags discarded.
Loading query file #2: chess3.0.1.gtf
  168451 query transfrags loaded.
  372 duplicate query transfrags discarded.
  1053854 reference transcripts loaded.
  214394 query transfrags loaded.
# gffcompare v0.12.6 | Command line was:
#gffcompare -r Primary_and_Assembled.combined.gtf devBrainIsoSeq_hg38.gtf -o IsoSeq_vs_Primary_and_Assembled
#

#= Summary for dataset: devBrainIsoSeq_hg38.gtf 
#     Query mRNAs :  214394 in   23376 loci  (206255 multi-exon transcripts)
#            (13767 multi-transcript loci, ~9.2 transcripts per locus)
# Reference mRNAs : 1053854 in  186194 loci  (924838 multi-exon)
# Super-loci w/ reference transcripts:    18217
#-----------------| Sensitivity | Precision  |
        Base level:    12.4     |    79.7    |
        Exon level:    14.3     |    64.7    |
      Intron level:    23.2     |    85.7    |
Intron chain level:     6.8     |    30.7   

* 66325 / 214394 of the IsoSeq transcripts (31%) match the combined CHESS Primary + Assembled Transcriptomes
* 148069  / 214394 IsoSeq transcripts are NOT found in CHESS Primary or Assembled GTFs

### 4. NOVEL_only IsoSeq vs (Primary + Assembled) Transcripts


In [26]:
gffcompare -r Primary_and_Assembled.combined.gtf devBrainIsoSeq_hg38_novel.gtf -o IsoSeqNovel_vs_Primary_and_Assembled
cat IsoSeqNovel_vs_Primary_and_Assembled.stats

  1053854 reference transcripts loaded.
  150038 query transfrags loaded.
# gffcompare v0.12.6 | Command line was:
#gffcompare -r Primary_and_Assembled.combined.gtf devBrainIsoSeq_hg38_novel.gtf -o IsoSeqNovel_vs_Primary_and_Assembled
#

#= Summary for dataset: devBrainIsoSeq_hg38_novel.gtf 
#     Query mRNAs :  150038 in   13162 loci  (149132 multi-exon transcripts)
#            (10128 multi-transcript loci, ~11.4 transcripts per locus)
# Reference mRNAs : 1053854 in  186194 loci  (924838 multi-exon)
# Super-loci w/ reference transcripts:    11107
#-----------------| Sensitivity | Precision  |
        Base level:     7.9     |    76.6    |
        Exon level:    10.1     |    65.3    |
      Intron level:    17.5     |    83.8    |
Intron chain level:     2.8     |    17.4    |
  Transcript level:     2.5     |    17.4    |
       Locus level:     4.4     |    66.2    |

     Matching intron chains:   25897
       Matching transcripts:   26084
              Matching loci:    8207

   

* 26084 / 150038 of NOVEL IsoSeq Transcripts (17.4%) are found in CHESS Primary + Assembled

e.g. 

## CONCLUSION
#### 124k (82.6%) of NOVEL IsoSeq Transcripts identified in the developing human brain are not present in CHESS (168k) or Assembled (987k) Transcriptomes